### ISSUE TITLE
feat(rpc): Implement starknet_traceBlockTransactions RPC call

### ISSUE THREAD

**Issue by Tbelleng** [keep-starknet-strange/madara#1372]
**feat(rpc): Implement starknet_traceBlockTransactions**

This PR aims to implement starknet_traceBlockTransactions rpc call introduced in v0.4.0-rc1 according to the v0.5.1 RPC specs described in https://github.com/keep-starknet-strange/madara/issues/1291

starknet_traceBlockTransactions

Retrieve traces for all transactions in the given block

**Arguments**
```
block_id : the hash of the requested block, or number (height) of the requested block, or a block tag
```

**Return**

Returns the execution traces of all transactions included in the given block

**Errors**

This function might return the following errors:
```
BLOCK_NOT_FOUND - The specified blocj could not be found.
```

**Comment by aniketpr01** [comment]
@Tbelleng can i take this?

**Comment by Tbelleng** [comment]
Yes, just check with @tdelabro before !

**Comment by aniketpr01** [comment]
@Tbelleng 
Checked with @tdelabro and he approved.
Just to understand context, need to implement this RPC call which returns execution traces of all transactions included in the given block, why we may encounter the error mentioned above?

**Comment by tdelabro** [comment]
The main concern is that I have no idea how you will get the execution traces.
I think you will have to re-execute the whole block in some way, while recording the trace somewhere.
What is your plan @aniketpr01 ?

**Comment by aniketpr01** [comment]
@tdelabro 
As discussed on TG, I can utilize `runtime_api()` to simulate and re-execute tx's in an environment of block preceding the one of interest (block n-1). 
This can enable me to trace and collect execution details of a block and implement this RPC call.

**Comment by tdelabro** [comment]
@aniketpr01 @Tbelleng who is working on this? I think I talked to @aniketpr01 on TG

**Comment by Tbelleng** [comment]
> yes @tdelabro, I will be working on it!

how its going ? @aniketpr01

**Comment by aniketpr01** [comment]
@Tbelleng 
I am still debugging simulating the transactions and requesting some more time on this!
Also could you please share your Telegram id or if you can DM me? TG: aniketpr01

**Comment by Tbelleng** [comment]
> @Tbelleng I am still debugging simulating the transactions and requesting some more time on this! Also could you please share your Telegram id or if you can DM me? TG: aniketpr01

Done !

**Comment by tdelabro** [comment]
closed by #1416
